src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -600,7 +606,16 @@ private IEnumerable<string> GetRestoreRuntimes(IEnumerable<string> projectRuntim
                         }
                         imports = runtimeSpec.Import;
                     }
-                    effectiveRuntimeSpecs.Add(runtimeSpec);
+
+                    allRuntimeNames.Add(runtimeSpec.Name);
+
+                    // Load dependencies provided by this runtime file
+                    foreach (var dependencySpec in runtimeSpec.Dependencies.Values)
+                    {
+                        // New dependency specs replace existing ones
+                        // This matches the NuGet3 behavior: https://github.com/NuGet/NuGet3/blob/dev/src/NuGet.RuntimeModel/RuntimeDescription.cs#L99
+                        effectiveRuntimeSpecs[dependencySpec.Name] = dependencySpec;
```

Comment:
After comparing with the NuGet3 code, this is the correct ordering (the original fix did the reverse)